packages/prosemirror-resizable-view/src/prosemirror-resizable-view.ts

```diff
@@ -1,172 +1,238 @@
   // size when `update` method is called.
-  width = '';
+  #width = '';
+  #height = '';
 
   constructor({
     node,
     view,
     getPos,
-    createElement,
+    aspectRatio = ResizableRatioType.Fixed,
+    options,
+    initialSize,
   }: {
     node: ProsemirrorNode;
     view: EditorView;
     getPos: () => number;
-    createElement: CreateElement;
+    aspectRatio?: ResizableRatioType;
+    options?: OptionShape;
+    initialSize?: { width: number; height: number };
   }) {
-    const outer = document.createElement('div');
-    outer.style.position = 'relative';
-    outer.style.width = node.attrs.width;
-    outer.style.maxWidth = '100%';
-    outer.style.minWidth = '50px';
-    outer.style.display = 'inline-block';
-    outer.style.lineHeight = '0'; // necessary so the bottom right handle is aligned nicely
-    outer.style.transition = 'width 0.15s ease-out, height 0.15s ease-out'; // make sure transition time is larger then mousemove event's throttle time
-
-    const inner = createElement({ node, view, getPos });
-
-    const handle = document.createElement('div');
-    handle.style.position = 'absolute';
-    handle.style.bottom = '0px';
-    handle.style.right = '0px';
-    handle.style.width = '16px';
-    handle.style.height = '16px';
-    handle.style.borderBottom = handle.style.borderRight = '4px solid #000';
-    handle.style.display = 'none';
-    handle.style.zIndex = '100';
-    handle.style.cursor = 'nwse-resize';
-
-    const setHandleDisplay = (visible?: true | string) => {
-      visible = visible || handle.dataset.mouseover || handle.dataset.dragging;
-      handle.style.display = visible ? '' : 'none';
-    };
+    const outer = this.createWrapper(node, initialSize);
+    const element = this.createElement({ node, view, getPos, options });
+
+    /**
+     * Only initialize `handleBottom`, `handleBottomRight`, and `handleBottomLeft` in
+     * `ResizableRatioType.Flexible` mode
+     */
+    const types =
+      aspectRatio === ResizableRatioType.Flexible
+        ? [
+            ResizableHandleType.Right,
+            ResizableHandleType.Left,
+            ResizableHandleType.Bottom,
+            ResizableHandleType.BottomRight,
+            ResizableHandleType.BottomLeft,
+          ]
+        : [ResizableHandleType.Right, ResizableHandleType.Left];
+
+    const handles = types.map((type) => new ResizableHandle(type));
+
+    for (const handle of handles) {
+      handle.dom.addEventListener('mousedown', (e) => this.resizeHandler(e, view, getPos, handle));
+      outer.append(handle.dom);
+    }
 
     outer.addEventListener('mouseover', () => {
-      handle.dataset.mouseover = 'true';
-      setHandleDisplay(true);
+      handles.forEach((handle) => handle.setHandleVisibility(true));
     });
 
     outer.addEventListener('mouseout', () => {
-      handle.dataset.mouseover = '';
-      setHandleDisplay();
+      handles.forEach((handle) => handle.setHandleVisibility(false));
     });
```

Comment:
Should we add a [`destroy`](https://prosemirror.net/docs/ref/#view.NodeView.destroy) handler, to remove these event listeners?